pkgs/by-name/xh/xhtml1/package.nix

```diff
@@ -1,33 +1,54 @@
-{ lib, stdenv, fetchurl, libxml2 }:
+{
+  lib,
+  fetchurl,
+  libxml2,
+  stdenv,
+}:
 
-stdenv.mkDerivation {
+stdenv.mkDerivation (finalAttrs: {
   pname = "xhtml1";
-  version = "unstable-2002-08-01";
+  version = "20020801";
 
-  src = fetchurl {
-    url = "https://www.w3.org/TR/xhtml1/xhtml1.tgz";
-    sha256 = "0rr0d89i0z75qvjbm8il93bippx09hbmjwy0y2sj44n9np69x3hl";
-  };
+  src =
+    let
+      year = builtins.substring 0 4 finalAttrs.version;
+    in
+    fetchurl {
+      url = "https://www.w3.org/TR/${year}/REC-xhtml1-${finalAttrs.version}/xhtml1.tgz";
+      hash = "sha256-FI6ezLXJEiK18MBzWRdMoN8b10g0orrkxuV8EBNqIGc=";
+    };
 
   nativeBuildInputs = [ libxml2 ];
 
-  installPhase =
+  installPhase = lib.concatStringsSep "\n" [
+    ''
+      runHook preInstall
     ''
-      mkdir -p $out/xml/dtd/xhtml1
-      cp DTD/*.ent DTD/*.dtd $out/xml/dtd/xhtml1
 
-      # Generate an XML catalog.
-      cat=$out/xml/dtd/xhtml1/catalog.xml
-      xmlcatalog --noout --create $cat
+    ''
+      mkdir -p $out/xml/dtd/xhtml1/
+      cp DTD/*.ent DTD/*.dtd $out/xml/dtd/xhtml1/
+    ''
+    # Generate XML catalog
+    ''
+      catalog=$out/xml/dtd/xhtml1/catalog.xml
+      xmlcatalog --noout --create $catalog
       grep PUBLIC DTD/*.soc | while read x; do
-          eval a=($x)
-          xmlcatalog --noout --add public "''${a[1]}" "''${a[2]}" $cat
+        eval a=($x)
```

Comment:
Nit: isn't shell indent 4 chars in nixpkgs?

Reply:
First hit for "indent" that I found in nixpkgs is .editorconfig, which lists 4 chars for *.sh files. (Yes, that's probably for *files*, not inline strings, but I think it's good to follow same style everywhere.) Not a big deal. I'm resolving this conversation.